### ISSUE TITLE
Frequency out of range when the format is floating point.

### ISSUE THREAD

**Issue by JiaxuanHan-UoS** [iicsys/pypmu#29]
Firstly, I set the format of phasors and frequency as integer. Everything is fine and I keep receiving messages about dataframe were received successfully.

When the format of frequency were changed to floating point. The error messages were shown below:
![image](https://user-images.githubusercontent.com/71255321/93076150-d2225900-f67e-11ea-9957-c621f4303bda.png)

According to the suggestions of Steven, I removed exception catching and find the error location is the range filter of frequency. Row 2143 in frame.py.
![image](https://user-images.githubusercontent.com/71255321/93076355-1a417b80-f67f-11ea-91fb-783d5d7cf600.png)

However, I checked the received data with PMU connection tester, and the received FREQ is within the given range(49.9288Hz).
![image](https://user-images.githubusercontent.com/71255321/93076423-39400d80-f67f-11ea-8b0f-8d47526cb59d.png)

Thanks for all discussion!

**Comment by poledna** [comment]
I also had the same issue a while back, and my "solution" was to comment out line 2144 and 2145 
in master
https://github.com/iicsys/pypmu/blob/66e6c495a91efb8a6018061ff7955a6654b1404d/synchrophasor/frame.py#L2144-L2145

Not a pretty solution, but it solved for me. As freq is a number bypassing the if should solve the issue.


in the branch udp-support-v0.1 there is already the commented version, but if you're using TCP i'd suggest keeping with master as it is more up to date with community issues.
https://github.com/iicsys/pypmu/blob/22b60642527c43a91513c5deb26cb8399b64f305/synchrophasor/frame.py#L2143-L2145

**Comment by JiaxuanHan-UoS** [comment]
@poledna 
Hi poledna, 
I tried your method and received the data frames. But I am pretty sure that my data is incorrect, the frequency is always floating between 50.5Hz. These are the curves of the actual frequency and the decoded frequency. 
![image](https://user-images.githubusercontent.com/71255321/93311997-96160200-f7fe-11ea-9e97-ec25b1fada72.png)

![image](https://user-images.githubusercontent.com/71255321/93311921-7da5e780-f7fe-11ea-9bda-cb1dcf6bb66b.png)

And the table in Mysql could also prove this.
![image](https://user-images.githubusercontent.com/71255321/93312752-7b905880-f7ff-11ea-9d97-6efc14b51469.png)

So did you received the correct data?

**Comment by sstevan** [comment]
Hey @JiaxuanHan-UoS,

Would you mind sharing your configuration frame with us?

Also, there is a known issue with FREQ unpacking caused by my misinterpretation as stated here #4 
Please try to switch to the `bugfix` branch and let us know if you are still having these issues.

**Comment by JiaxuanHan-UoS** [comment]
@sstevan 

Hi Stevan,

I just read [issue4](https://github.com/iicsys/pypmu/issues/4) and find there could be some misunderstanding about FREQ definition. If the format of FREQ is integer, the 2-byte string represents deviation and requires the nominal frequency defined in CONFIG-2 to determine the actual frequency. However, if the FREQ is float, the 4-byte string is long enough to represent the accurate value and don't need the nominal frequency in CONFIG-2 to calculate the result. 
![image](https://user-images.githubusercontent.com/71255321/93316444-1854f500-f804-11ea-9412-1e45a19d39e7.png)
In the above picture, red part means the FREQ in float format (and blue for DFREQ). If we transfer it from hex to decimal:
![image](https://user-images.githubusercontent.com/71255321/93316598-54885580-f804-11ea-971f-ccde963359af.png)
The result is exactly 50.168

**Comment by sstevan** [comment]
That is correct. That issue has been already fixed on bugfix branch.
Have you checked the ConfigurationFrame?

If you are using tinyPDC this line will ask for ConfigurationFrame in order to unpack the values:

```config = pdc.get_config()  # Get configuration from PMU```

Please double check your FORMAT field in configuration frame, should look like this:

```
Data format in data frames, 16-bit flag.
Bits 15–4: Unused
Bit 3: 0 = FREQ/DFREQ 16-bit integer, 1 = floating point
Bit 2: 0 = analogs 16-bit integer, 1 = floating point
Bit 1: 0 = phasors 16-bit integer, 1 = floating point
Bit 0: 0 = phasor real and imaginary (rectangular), 1 = magnitude and angle (polar)
```

So you should have ```Bit 3 = 1```.

**Comment by JiaxuanHan-UoS** [comment]
@sstevan 
![image](https://user-images.githubusercontent.com/71255321/93327556-6cff6c80-f812-11ea-99fb-521127e0435a.png)
Yes, this is the ConfigurationFrame of my PMU. The format is 0x0009, bit 3 and bit 0 = 1.
![image](https://user-images.githubusercontent.com/71255321/93327776-bcde3380-f812-11ea-8ce6-0752e871170e.png)
Just the same with the openPDC configuration.

**Comment by JiaxuanHan-UoS** [comment]
Thanks, the problem has been solved after using bug-fix v1.0.1 frame.py [scripts](https://github.com/iicsys/pypmu/blob/bugfix-v1.0.1/synchrophasor/frame.py)
